### DISCUSSION TITLE
Allow to use ref struct as an argument to a dynamically dispatched operation

### DISCUSSION THREAD

**Discussion by TSlivede** [dotnet/csharplang#5271] [Language Ideas]
Would it be hard to allow the use of values of ref struct type as arguments to a dynamically dispatched operations?

Currently an error like
> Cannot use an expression of type 'MyRefStruct' as an argument to a dynamically dispatched operation.
is given by the compiler.

(see https://dotnetfiddle.net/d0B7vm)

If this was possible it would simplify calling methods with ref struct arguments using reflection.

The error seems to be `ERR_BadDynamicMethodArg`. If I skimmed over the code correctly it's from here:

https://github.com/dotnet/roslyn/blob/b9cce7fa5144907c7ceb60dc4a9bda0e8cd0069d/src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs#L903-L909

Sadly there is no comment above, that would explain, why this is not possible...

https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/types/using-type-dynamic also doesn't mention anything about ref structs...

**Comment by 333fred** [comment]
Dynamic involves reflection and boxing all arguments. Ref structs cannot be boxed. Additionally, we cannot determine the appropriate lifetime rules in the presence of dynamic to ensure stack references never escape.
